database/migrations/000029_bundle_subscription.up.sql

```diff
@@ -0,0 +1,43 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+CREATE TABLE bundles(
+    -- I have given this a separate PK to simplify some of the queries
+    -- the combination of the remaining columns are unique
+    id          UUID NOT NULL DEFAULT gen_random_uuid() PRIMARY KEY,
+    namespace   TEXT NOT NULL,
+    name        TEXT NOT NULL,
+    UNIQUE      (namespace, name)
+);
+
+CREATE TABLE subscriptions(
+    -- same comment as for PK of `bundles`
+    id                UUID NOT NULL DEFAULT gen_random_uuid() PRIMARY KEY,
+    project_id        UUID NOT NULL REFERENCES projects(id) ON DELETE CASCADE,
+    -- this FK is not ON DELETE CASCADE
+    -- this prevents deletion of bundles which are still in use
```

Comment:
uhm... wait, what would happen is that if a bundle gets deleted and a subscription is still around, the bundle will get deleted, but bundles that used to use that subscription will not be. I think we want the `ON DELETE CASCADE` in that case.

What we would want though is a trigger or something else to check that we don't delete a bundle if it's still in use. right?

Reply:
As per the chat on Slack - my plan is to have code which will clean up unused bundles when subscriptions are deleted or updated. I am open to considering some kind of DB driven cleanup, but I generally prefer to do it explicitly in code.